packages/lexicon/src/validators/formats.ts

```diff
@@ -0,0 +1,107 @@
+import { ensureValidAtUri } from '@atproto/uri'
+import { isValidISODateString } from 'iso-datestring-validator'
+import { CID } from 'multiformats/cid'
+import { ValidationResult, ValidationError } from '../types'
+import { ensureValidDid, ensureValidHandle } from '@atproto/identifier'
+import { ensureValidNsid } from '@atproto/nsid'
+
+export function datetime(path: string, value: string): ValidationResult {
+  try {
+    if (!isValidISODateString(value)) {
+      throw new Error()
+    }
+  } catch {
+    return {
+      success: false,
+      error: new ValidationError(
+        `${path} must be an iso8601 formatted datetime`,
+      ),
+    }
+  }
+  return { success: true, value }
+}
+
+export function uri(path: string, value: string): ValidationResult {
+  const isUri = value.match(/\w+:(\/?\/?)[^\s]+/) !== null
```

Comment:
Since `[^\s]+` covers `\/?\/?`, I am not sure the `(\/?\/?)` is affecting the result.  But I also see why it's there / what it's getting at.  Just to toss a close alternative out there:
```suggestion
  const isUri = value.match(/\w+:(?:\/\/)?[^\s\/][^\s]*/) !== null
```

If we're not 100% sure how we want it, I think it's also totally cool to leave it as-is and come back to it later.

Reply:
nice im game 👌